### DISCUSSION TITLE
Problems with `null` and `DbNull`

### DISCUSSION THREAD

**Discussion by MorenoMdz** [prisma/prisma#21833] [Q&A]
### Question

We are finalizing a big NoSQL to SQL (PG) migration using Prisma, and in a few cases we had to use `Json` in our schema, but then when we are writing the data in `update` or `upsert` calls, we endup with type errors such:

Model:

```prisma
// Model
breaks                       Json?
```

Error:

```
// Type error
...not assignable to type 'ModelUpdateInput'.
      Types of property 'breaks' are incompatible.
        Type 'JsonArray | null' is not assignable to type 'NullableJsonNullValueInput | InputJsonValue | undefined'.
          Type 'null' is not assignable to type 'NullableJsonNullValueInput | InputJsonValue | undefined'.ts(2322)
```

If we use `DbNull` it will correctly mark the column value as `NULL` in the DB. For example
`breaks: update.breaks !== null ? update.breaks : Prisma.DbNull,`

If we use `Prisma.JsonNull` we get a weird string like `null` value
`breaks: breaks !== null ? (breaks as Prisma.JsonArray) : Prisma.JsonNull,` 

And, if we do simply try to mark one of those fields to `null`, it will store as the string `null` value, for example:

```ts
    return prismaClient.model.upsert({
      where: {
        id
      },
      create: {
        ...data,
        breaks: null,
      },
      update: data,
    })
  }
```

<img width="114" alt="image" src="https://github.com/prisma/prisma/assets/29472480/d9be2f46-49b4-47c4-9a9b-9dc6cd58cf4a">

This is a problem as my queries using `IS NULL` do not find those `JsonNull` typed values.

I will need to do a big pass and search for those string like `null` values and clean everything, but

**Comment by ludralph** [comment]
Hi @MorenoMdz 👋 

Thank you for raising this question.

Can you please clarify how you are performing the query using `IS NULL` with an example?

If you need to filter a null value using JSON, we generally recommend using `AnyNull` as shown in the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-by-null-values).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
